frontend/components/site/SiteSettings/ScanConfigs.jsx

```diff
@@ -0,0 +1,271 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import api from '../../../util/federalistApi';
+import { BUILD_SCAN_RULES } from '../../../util/scanConfig';
+import { useDefaultScanRules } from '../../../hooks/useDefaultScanRules';
+import { useSiteBuildTasks } from '../../../hooks/useSiteBuildTasks';
+import notificationActions from '../../../actions/notificationActions';
+import ExpandableArea from '../../ExpandableArea';
+import { IconTrash, IconExternalLink } from '../../icons';
+import LoadingIndicator from '../../LoadingIndicator';
+
+function getRuleName(rule) {
+  return BUILD_SCAN_RULES.find(r => r.id === rule.id)?.name;
+}
+function getRuleLink(rule) {
+  return BUILD_SCAN_RULES.find(r => r.id === rule.id)?.url;
+}
+function ScanConfigs({ siteId: id }) {
+  // TODO: maybe someday this needs to take siteBuildTask.branch into account
+
+  const { siteBuildTasks } = useSiteBuildTasks(id);
+  const [isLoading, setIsLoading] = useState(false);
+
+  // we use default scan rules
+  // and customer scan rules from siteBuildTasks
+  const { defaultScanRules } = useDefaultScanRules();
+  const [customerRules, setCustomerRules] = useState([]);
+
+  useEffect(() => {
+    setCustomerRules(
+      siteBuildTasks.map(sbt => (sbt.metadata?.rules ? sbt.metadata.rules : [])).flat()
+    );
+  }, [siteBuildTasks]);
+
+  function handleUpdate(siteBuildTask) {
+    const typeRules = customerRules
+      .filter(rule => rule.type === siteBuildTask.id)
+      .filter(rule => rule.id.slice(0, 4) !== 'temp'); // don't save temp rules
+
+    return api.updateSiteBuildTask(id, siteBuildTask.sbtId, { rules: typeRules })
+      .then(() => notificationActions.success('Successfully saved scan configuration.'))
+      .catch(() => notificationActions.success('Error saving scan configuration.'))
+      .finally(() => setIsLoading(false));
```

Comment:
The `Save all rules` button is always enabled no matter if there are no changes or the config is being updated. Should we expand the use of useState to disable the button to add additional context to the user?